### PR TITLE
[Merged by Bors] - chore: bump Std dependency

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Basic.lean [hunk 1]
```diff
@@ -690,36 +690,23 @@ alias ⟨_, not_exists_of_forall_not⟩ := exists_imp
...
-protected theorem Decidable.not_exists_not [∀ x, Decidable (p x)] : (¬∃ x, ¬p x) ↔ ∀ x, p x := by
-  simp only [not_exists, Decidable.not_not]
 #align decidable.not_exists_not Decidable.not_exists_not
 
 theorem not_exists_not : (¬∃ x, ¬p x) ↔ ∀ x, p x := Decidable.not_exists_not
 #align not_exists_not not_exists_not
 
 lemma forall_or_exists_not (P : α → Prop) : (∀ a, P a) ∨ ∃ a, ¬ P a := by
-  rw [← not_forall]; exact em _
+  rw [← Classical.not_forall]; exact em _
```
Comment (digama0) [inline Mathlib/Logic/Basic.lean:709]: Hm, this seems like a regression. Does `export` not work properly? If so, you should try using `alias` instead.

Comment (semorrison) [reply Mathlib/Logic/Basic.lean:709]: Oops, my mistake. I did that one before switching to using `export`.

Comment (digama0) [reply Mathlib/Logic/Basic.lean]: there are a few others below

File: Mathlib/Logic/Basic.lean [hunk 2]
```diff
@@ -1186,12 +1163,12 @@ theorem ite_eq_iff' : ite P a b = c ↔ (P → a = c) ∧ (¬P → b = c) := dit
 #align ite_eq_right_iff ite_eq_right_iff
 
 theorem dite_ne_left_iff : dite P (fun _ ↦ a) B ≠ a ↔ ∃ h, a ≠ B h := by
-  rw [Ne.def, dite_eq_left_iff, not_forall]
+  rw [Ne.def, dite_eq_left_iff, Classical.not_forall]
```
Comment (digama0) [inline Mathlib/Logic/Basic.lean:1166]:
```suggestion
  rw [Ne.def, dite_eq_left_iff, not_forall]
```

File: Mathlib/Logic/Basic.lean [hunk 3]
```diff
@@ -1186,12 +1163,12 @@ theorem ite_eq_iff' : ite P a b = c ↔ (P → a = c) ∧ (¬P → b = c) := dit
...
   exact exists_congr fun h ↦ by rw [ne_comm]
 #align dite_ne_left_iff dite_ne_left_iff
 
 theorem dite_ne_right_iff : (dite P A fun _ ↦ b) ≠ b ↔ ∃ h, A h ≠ b := by
-  simp only [Ne.def, dite_eq_right_iff, not_forall]
+  simp only [Ne.def, dite_eq_right_iff, Classical.not_forall]
```
Comment (digama0) [inline Mathlib/Logic/Basic.lean:1171]:
```suggestion
  simp only [Ne.def, dite_eq_right_iff, not_forall]
```

File: lake-manifest.json [hunk 4]
```diff
@@ -25,20 +25,20 @@
...
     "name": "aesop",
     "inputRev?": "master",
+    "inherited": false}},
+  {"git":
+   {"url": "https://github.com/leanprover/std4",
+    "subDir?": null,
+    "rev": "0d90d74628daa9ee7d4039a3b035240ff6d64da5",
+    "opts": {},
+    "name": "std",
+    "inputRev?": "main",
```
Comment (eric-wieser) [inline lake-manifest.json:42]: I guess the lake stability bug is still going...

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

File: lake-manifest.json (hunk 4 above)
Comment (semorrison) [reply lake-manifest.json:42]: Yes, it is tracked at leanprover/lean4#2664.

**Pull request closed without merging**